src/tracing/vizceral_json_file_creator_nodes.coffee

```diff
@@ -0,0 +1,176 @@
+      reducedMap = reduceMap traceIdToTopicChannel
+      srcToTarget = getSourceToTarget reducedMap
+      entryhead = findEntryHeads srcToTarget
+      nodeList = getNodeList srcToTarget
+      srcToTargetList = getSourceToTargetList srcToTarget, entryhead, mapOfTopicToCount
+      output = prepareJSON nodeList, srcToTargetList
+      callback null, JSON.stringify output
+
+  # this function filters data and error data in two different lists based on the length of incoming data
+  filterDataAndError = (allData) ->
+    ###
+    Two kind of data 1. success 2. error
+    1.<Tsource:Csource|Tdestination| rev-id-| Date.now() to keep it unique
+    2.<rev-id| Tdestination:Cdestination | Date.now()>
+    ###
+    errors = []
+    success = []
+    for key, value of allData
+      {length} = value.split('|')
+      if length is 3 then errors.push value else success.push value
+    [success, errors]
+
+  # this function gets the no of time error occured and creates a map based on that
+  getErrorMapOfTopicToCount = (errorData)->
+    errorNodeMap = {}
+    for key, value in errorData
+      node = key.split('|')[1].split(':')[0]
+      errorNodeMap[node] or= 0
+      errorNodeMap[node]++
+    errorNodeMap
+
+  # This function creates a  map of key(traceId=revNo + Date.Now()) to the topic channel communication
+  getTopicToChannelbyId = (success) ->
+    traceIdToTopicChannel = {}
+    #for value, index in success
+    for key, value in success
+      dataArray = key.split('|')[0..1].join('>')
+      key_trace = key.split('|')[2]
+      if key_trace of traceIdToTopicChannel
+        traceIdToTopicChannel[key_trace] = traceIdToTopicChannel[key_trace] + "," + dataArray
+      else
+        traceIdToTopicChannel[key_trace] = dataArray
+    traceIdToTopicChannel
+
+  # This function reduces the map by bringing in count for repetition
+  reduceMap = (traceIdToTopicChannel) ->
+    reducedMap = {}
+    for key, value of traceIdToTopicChannel
+      arrayOfTC = value.split(',')
+      for item, index in arrayOfTC
+        if arrayOfTC[index] of reducedMap
+          reducedMap[item]++
+        else
+          reducedMap[item] = 1
+    reducedMap
+
+  # This function prepares output list src>targer>#count
+  getSourceToTarget = (reducedMap) ->
+    size = _.keys(reducedMap).length
+    outputList = []
+    for value, index of reducedMap
+      [src, trg] = value.split('>')
+      source = src.split(':')[0]
+      target = trg
+      outputList.push(source+'>'+target+'#'+reducedMap[value])
+    outputList
+
+  # Ths function prepares the nodeList for adding to JSON finally
+  getNodeList = (srcToTarget) ->
+    resultNodes = []
+    resultNodes.push('narrows')
+    for row in srcToTarget
+      [first, rest] = row.split('>')
+      if first not in resultNodes
+        resultNodes.push(first)
+      if rest.split('#')[0] not in resultNodes
+        resultNodes.push(rest.split('#')[0])
+    for node in resultNodes
```

Comment:
I would add an empty newline above this help visually separate the return value.